### PR TITLE
Enhance modal form redirects

### PR DESCRIPTION
**Pull request by yhabteab**

### CODE REVIEW

**Comment by lippserd** [comment]
@yhabteab This is not ready, right?

**Comment by yhabteab** [comment]
> @yhabteab This is not ready, right?

It's ready, but its kind of waiting for https://github.com/Icinga/icingaweb2/pull/5068 to be merged.

File: application/controllers/ReportController.php [hunk 1]
```diff
@@ -49,14 +51,29 @@ public function indexAction()
...
+        $this->controls->getAttributes()->add('class', 'default-layout');
         $this->addControl($this->assembleActions());
 
+        $contentId = $this->content->getAttributes()->get('id')->getValue();
+        $this->sendExtraUpdates([
+            $contentId => Url::fromPath('reporting/report/content', ['id' => $this->report->getId()])
+        ]);
+
+        // Will be replaced once the report content is rendered
+        $this->addContent(new HtmlElement('p'));
```
Comment (nilmerg) [inline application/controllers/ReportController.php:63]: Consider to use a element without a "meaning". I.e. `HtmlString::create('&nbsp;')` or `div`.

File: application/controllers/ReportController.php [hunk 2]
```diff
@@ -49,14 +51,29 @@ public function indexAction()
...
 
+        $this->view->compact = true;
+        $this->_helper->layout()->disableLayout();
+
         try {
-            $this->addContent($this->report->toHtml());
-        } catch (\Exception $e) {
+            $this->getDocument()->addHtml($this->report->toHtml());
+        } catch (Exception $e) {
             $this->addContent(Error::show($e));
```
Comment (nilmerg) [inline application/controllers/ReportController.php:77]: An error is also shown with tabs I suppose? Should also be added to the document directly.

File: application/controllers/ReportsController.php [hunk 3]
```diff
@@ -127,14 +117,27 @@ public function newAction()
...
+                Notification::success($this->translate('Created report successfully'));
+
+                $url = Url::fromPath('reporting/reports');
+                if ($form->getPressedSubmitElement()->getName() === 'create_show') {
+                    $url = Url::fromPath('reporting/report', ['id' => $form->getId()]);
+                } elseif ($class !== null) {
+                    $url = '__CLOSE__';
+
+                    // This will close only the modal view and leave the object detail view of Icinga DB Web untouched.
+                    $this
```
Comment (nilmerg) [inline application/controllers/ReportsController.php:135]: Don't break here pls, looks weird. You continue calling methods on the response. Not on the controller.

**Review by nilmerg — CHANGES_REQUESTED** [review]
* the export dropdown isn't usable while the report is loading. (Fixed by https://github.com/Icinga/ipl-web/pull/188)
* Apply `outline: none` to `.content.container:focus`.

Sidenote:

the tabs should use `data-base-target=_main`. Otherwise the detail keeps open while switching.

**Review by nilmerg — APPROVED** [review]
LGTM! The referenced Web PR is merged (kind of)

**Comment by lippserd** [comment]
I don't know if it's related to the framework or the changes here, but as soon as I remove a report, the report list is refreshed twice. The clone action also triggers a refresh, which is not required.

**Comment by yhabteab** [comment]
> I don't know if it's related to the framework or the changes here, but as soon as I remove a report, the report list is refreshed twice.

It's a framework bug.

> The clone action also triggers a refresh, which is not required.

By "triggers a refresh" do you mean that the `col2` container is updated after the redirection or that the `col1` container is refreshed as well?

File: phpstan-baseline.neon [hunk 4]
```diff
@@ -167,7 +167,7 @@ parameters:
 
 		-
 			message: "#^Cannot call method format\\(\\) on mixed\\.$#"
-			count: 2
+			count: 3
```
Comment (nilmerg) [inline phpstan-baseline.neon:170]: We have a solution for this, so please fix this properly.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Comment by nilmerg** [comment]
Shouldn't the cloned report be visible in the detail, instead of the source report?

**Comment by yhabteab** [comment]
> Shouldn't the cloned report be visible in the detail, instead of the source report?

No clue 🤷!

**Comment by nilmerg** [comment]
I'd say so. At least does the refresh of col2 make no sense currently, as nothing changed. And if I clone a report and want to schedule/send it afterwards, I don't have to search for it in the list.

File: application/controllers/ReportController.php [hunk 5]
```diff
@@ -89,8 +107,12 @@ public function cloneAction()
             ->setSubmitButtonLabel($this->translate('Clone Report'))
             ->setAction((string) Url::fromRequest())
             ->populate($values)
-            ->on(ReportForm::ON_SUCCESS, function () {
-                $this->redirectNow('__CLOSE__');
+            ->on(ReportForm::ON_SUCCESS, function (ReportForm $form) {
+                Notification::success($this->translate('Cloned report successfully'));
+
+                $this->closeModalAndRefreshRemainingViews(
```
Comment (nilmerg) [inline application/controllers/ReportController.php:113]: This works differently when on Icinga Web v2.12. Just use redirectNow.

**Review by nilmerg — CHANGES_REQUESTED** [review]
Please fix the conflict

**Review by nilmerg — APPROVED** [review]

**Review by nilmerg — CHANGES_REQUESTED** [review]
Oh no, wait. The time frame items still open a modal.

**Comment by nilmerg** [comment]
Never mind. But please use `openInModal` where applicable. (As you noted by yourself)

**Review by nilmerg — APPROVED** [review]

**Pull request merged**